third_party/amd/lib/TritonAMDGPUTransforms/StreamPipeline.cpp

```diff
@@ -484,6 +485,11 @@ findPipelineableLoads(scf::ForOp forOp,
   DenseSet<Operation *> seen;
   // Recursively visit the given op and its operands to discover all load ops
   // and collect their distances and uses.
+
+  auto arch = getAMDArch(forOp->getParentOfType<ModuleOp>());
+  std::optional<llvm::AMDGPU::GPUKind> gpuKind = std::nullopt;
+  if (arch)
+    gpuKind = llvm::AMDGPU::parseArchAMDGCN(*arch);
```

Comment:
You can use `deduceISAFamily(*arch)` to avoid repeating the logic.

Reply:
I wasn't sure if the association with CDNA4 was correct but ok sure I can do that

Reply:
Yeah it's CDNA4 there.

Reply:
Done